apps/accelerate/forward_forward/forward_forward/operations/data.py

```diff
@@ -0,0 +1,148 @@
+import urllib.request
+from typing import Any
+
+import torch
+import torch.utils.data
+from torchvision import datasets, transforms
+
+from nebullvm.operations.base import Operation
+
+
+class MNISTDataLoaderOperation(Operation):
+    """DataLoaderOperation"""
+
+    def get_result(self) -> Any:
+        return self.state.get("train_data"), self.state.get("test_data")
+
+    def execute(self, batch_size: int, shuffle: bool):
+        train_loader = torch.utils.data.DataLoader(
+            datasets.MNIST(
+                "data",
+                train=True,
+                download=True,
+                transform=transforms.Compose(
+                    [
+                        transforms.ToTensor(),
+                        transforms.Normalize((0.1307,), (0.3081,)),
+                    ]
+                ),
+            ),
+            batch_size=batch_size,
+            shuffle=shuffle,
+        )
+        test_loader = torch.utils.data.DataLoader(
+            datasets.MNIST(
+                "data",
+                train=False,
+                transform=transforms.Compose(
+                    [
+                        transforms.ToTensor(),
+                        transforms.Normalize((0.1307,), (0.3081,)),
+                    ]
+                ),
+            ),
+            batch_size=1000,
+            shuffle=False,
+        )
+        self.state["train_data"] = train_loader
```

Comment:
This is a general comment for whenever we save the result of an operation: to be consistent with the speedster app, we should save each result in a separate attribute instead of defining a dictionary state. So for example in this case we would have self.train_data and self.test_data.

Reply:
fixed in https://github.com/nebuly-ai/nebullvm/pull/126/commits/914dfbfd697521dac82df6fed05a3820b2950f18